### PR TITLE
Add offset parameter to trade aggregations

### LINKED ISSUE #545: Trade aggregation: add offset parameter
Currently, trade aggregation buckets are aligned with epoch. This presents a challenge with aligning daily and weekly aggregations to a specific timezone.

### PR DESCRIPTION
**Pull request by lechengfan**
- Closes #545 
- Adds an offset parameter to TradeAggregationsQ and a WithOffset method.
- Change actions_trade.go so that an offset filter can be created and used.
- Add test

### CODE REVIEW

File: services/horizon/internal/actions_trade.go [hunk 1]
```diff
@@ -147,6 +147,7 @@ type TradeAggregateIndexAction struct {
 	Action
 	BaseAssetFilter    xdr.Asset
 	CounterAssetFilter xdr.Asset
+	OffsetFilter       time.Millis
```
Comment (tomerweller) [inline services/horizon/internal/actions_trade.go]: `time.Millis` is a timestamp but the offset is a time range. there's no functional difference but I would use an int64.

File: services/horizon/internal/db2/history/trade_aggregation.go [hunk 2]
```diff
@@ -101,7 +108,7 @@ func (q *TradeAggregationsQ) GetSql() sq.SelectBuilder {
 		Where(sq.Eq{"base_asset_id": q.baseAssetID, "counter_asset_id": q.counterAssetID})
 
 	//adjust time range and apply time filters
-	bucketSQL = bucketSQL.Where(sq.GtOrEq{"ledger_closed_at": q.startTime.ToTime()})
+	bucketSQL = bucketSQL.Where(sq.GtOrEq{"ledger_closed_at": (q.startTime + q.offset).ToTime()})
```
Comment (tomerweller) [inline services/horizon/internal/db2/history/trade_aggregation.go]: Looks like you're offsetting the start time for the entire range but the issue is about offsetting individual buckets

File: services/horizon/internal/actions_trade.go [hunk 3]
```diff
@@ -149,6 +149,7 @@ type TradeAggregateIndexAction struct {
 	CounterAssetFilter xdr.Asset
 	StartTimeFilter    time.Millis
 	EndTimeFilter      time.Millis
+	OffsetFilter       int64
```
[the remark this answers is not in the document]
Comment (lechengfan) [reply services/horizon/internal/actions_trade.go]: Yes, I've run gofmt on all of these files. What's the formatting issue?

File: services/horizon/internal/actions_trade_test.go [hunk 4]
```diff
@@ -169,19 +170,22 @@ func testTradeAggregationPrices(t *HTTPT, record horizon.TradeAggregation) {
 	testPrice(t, record.Close, record.CloseR)
 }
 
+const second = int64(1000)
+const minute = int64(60 * second)
+const hour = int64(minute * 60)
```
[the remark this answers is not in the document]
Comment (lechengfan) [reply services/horizon/internal/actions_trade_test.go]: Got it will make those changes

**Pull request merged**